lib/awful/screen.lua.in

```diff
@@ -104,6 +104,12 @@ function screen.padding(_screen, padding)
     return data.padding[_screen]
 end
 
+--- Get the focused screen.
+-- This can be replaced in a user's config.
+function screen.focused()
+    return capi.mouse.screen
```

Comment:
IIRC it was my initial approach also (probably in an amended commit), but then @psychon said that it should be kept as simple / predictable as possible.
And as you can see from the diff, it would reflect the current behavior, where only mouse.screen is being  considered in general.

But this is something that could be easily changed / improved then centrally.